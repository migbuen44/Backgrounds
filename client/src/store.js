import { configureStore } from '@reduxjs/toolkit';
import userLoginReducer from './slices/userLoginSlice';
import loginModalReducer from './slices/loginModalSlice';
import userInfoReducer from './slices/userInfoSlice';

export default configureStore({
  reducer: {
    userLogin: userLoginReducer,
    loginModal: loginModalReducer,
    userInfo: userInfoReducer,
  },
});
