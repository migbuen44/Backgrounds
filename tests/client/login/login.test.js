import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer'
import store from '../../../client/src/store';
import Login from '../../../client/src/components/login/login';

describe('Login', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Provider store={store}>
        <Login />
      </Provider>, div)
  })

  it('matches snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Login />
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  })
})

