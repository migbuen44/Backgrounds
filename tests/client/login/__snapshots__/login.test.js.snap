// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login matches snapshot 1`] = `
<div
  className="lib-login-module-userInfo-1ziX"
>
  <h2
    className="lib-login-module-header-3GFZ"
  >
    Login
  </h2>
  <form
    onSubmit={[Function]}
  >
    <div>
      <label>
        Email 
      </label>
      <input
        className="lib-login-module-email-2xXE"
        onChange={[Function]}
        type="email"
      />
    </div>
    <div>
      <label>
        Password 
      </label>
      <input
        className="lib-login-module-password-1jMN"
        onChange={[Function]}
        type="password"
      />
    </div>
    <button
      className="lib-login-module-submitLogin-kUim"
      onClick={[Function]}
      type="submit"
    >
      Login
    </button>
    <button
      className="lib-login-module-toSignUp-3v1K"
      onClick={[Function]}
      type="button"
    >
      SignUp
    </button>
  </form>
</div>
`;
